src/Compilers/CSharp/Test/Semantic/Semantics/RefLocalsAndReturnsTests.cs

```diff
@@ -903,15 +903,9 @@ public static void Main()
                 // (16,54): error CS8175: Cannot use ref local 'r' inside an anonymous method, lambda expression, or query expression
                 //         ref char Moo1(ref char a, ref char b) => ref r;
                 Diagnostic(ErrorCode.ERR_AnonDelegateCantUseLocal, "r").WithArguments("r").WithLocation(16, 54),
-                // (16,54): error CS8151: The return expression must be of type 'char' because this method returns by reference
-                //         ref char Moo1(ref char a, ref char b) => ref r;
-                Diagnostic(ErrorCode.ERR_RefReturnMustHaveIdentityConversion, "r").WithArguments("char").WithLocation(16, 54),
```

Comment:
Why is this error no longer applicable?

Reply:
We now mark the node as bad as soon as we find that "r" is not ref-returnable, no matter what its type is. The second error was unnecessary - fixing that would not help.

---
In reply to: [135400165](https://github.com/dotnet/roslyn/pull/21334#discussion_r135400165) [](ancestors = 135400165)